odl/space/cu_ntuples.py

```diff
@@ -847,36 +976,271 @@ def _norm_default(x):
     return x.data.norm()
 
 
+def _pnorm_default(x, p):
+    if p == float('inf'):
+        raise NotImplementedError('inf-norm not implemented.')
+    elif int(p) != p:
+        raise NotImplementedError('non-integer powers not implemented')
+    # TODO: optimized version in C++ code?
```

Comment:
Sorry for committing :-)
I should have used an exclamation mark instead. However, it will take me a while to find out how to do this. I'll make an issue in odlpp.